content/post/smaller-the-parameter-smaller-the-bugs.md

```diff
@@ -0,0 +1,243 @@
+---
+title: "Smaller the Parameter, Smaller the Bugs"
+date: 2021-07-13T22:57:39+09:00
+hero: /images/hero-5.jpg
+excerpt: "Some tips for coding efficiently and reducing bugs"
+timeToRead: 10
+authors:
+  - Bater
+draft: true
+---
+When I review code, I sometimes notice that there is room to squash parameters of methods, yet, not many developers are aware of this when they design code. From my coding experience, we can write code efficiently and reduce bug rates by putting in effort in reducing the parameters from our methods. Unless your manager still evaluates output by lines of code, you should definitely try these tips in your daily coding life.
+
+So today, I will share how to do it with OOP (object-oriented programming) design, and explain where this magic power comes from.
+
+Let's start with a small example. The following demo code was written in Ruby, but don't worry; Ruby code is so beginner-friendly that once you have some OOP background, it's easy to grasp.
+
+The example story is, we have some books and want to pass the author name as a parameter, and get the book title, author name, and publish date as a summary.
+```ruby
+class Book
+  attr_reader :title, :publish_year
+ 
+  def initialize(title, publish_year)
+    @title = title
+    @publish_year = publish_year
+  end
+ 
+  def summary(first_name, last_name)
+    "#{title} - #{last_name}, #{first_name} (#{publish_year})"
+  end
+end
+ 
+book = Book.new("Ruby is the best", "2020")
+book.summary("Bater", "Chen")
+=> "Ruby is the best - Chen, Bater (2020)"
+```
+We have a book class, it has two attributes, title, and publish_year. The `initialize` method would be called when we `new` a book object. Book class only has one instance method called `summary`, it needs two parameters, first_name and last_name. In Ruby, when we define the `attr_reader` as the attribute of the object, it will create the read method automatically. So we can call the instance variable directly by `title` to get `@title` return, same as publish_year.
```

Comment:
Maybe "title" and "publish_year" could be formatted with enclosing \`\`? e.g.  it has two attributes: `title` and `publish_year`. I see a bunch of other places where you're referencing classes and object names without \`\`